File: backend/src/main/java/com/taskapp/backend/service/TaskService.java
package com.taskapp.backend.service;

import com.taskapp.backend.model.Task;
import com.taskapp.backend.repository.TaskRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

@Service
public class TaskService {
    
    @Autowired
    private TaskRepository taskRepository;
    
    // Get all tasks
    public List<Task> getAllTasks() {
        return taskRepository.findAll();
    }
    
    // Get task by ID
    public Optional<Task> getTaskById(Long id) {
        return taskRepository.findById(id);
    }
    
    // Create new task
    public Task createTask(Task task) {
        return taskRepository.save(task);
    }
    
    // Update task
    public Task updateTask(Long id, Task taskDetails) {
        Task task = taskRepository.findById(id)
            .orElseThrow(() -> new RuntimeException("Task not found with id: " + id));
        
        task.setTitle(taskDetails.getTitle());
        task.setDescription(taskDetails.getDescription());
        task.setStatus(taskDetails.getStatus());
        task.setUpdatedAt(LocalDateTime.now());
        
        return taskRepository.save(task);
    }
    
    // Delete task
    public void deleteTask(Long id) {
        taskRepository.deleteById(id);
    }
}

File: frontend/src/services/taskService.js
import axios from "axios";

const API_BASE_URL =
  import.meta.env.VITE_API_URL || "http://localhost:8080/api";
const API_URL = `${API_BASE_URL}/tasks`;

export default {
  // Get all tasks
  getAllTasks() {
    return axios.get(API_URL);
  },

  // Get task by ID
  getTaskById(id) {
    return axios.get(`${API_URL}/${id}`);
  },

  // Create new task
  createTask(task) {
    return axios.post(API_URL, task);
  },

  // Update task
  updateTask(id, task) {
    return axios.put(`${API_URL}/${id}`, task);
  },

  // Delete task
  deleteTask(id) {
    return axios.delete(`${API_URL}/${id}`);
  },
};
